index.html

```diff
@@ -3004,109 +3004,347 @@ <h3>Truncating or limiting the length of strings</h3>
-		<p>First, the number of characters in the truncated string decreases as the number of bytes required per character goes up. So the Cyrillic string has half the number of characters as the ASCII string. The Chinese string has about 1/3 the number. And the emoji string has 1/4.</p>
-		<p>Second, in two of the three examples, the text is truncated on a byte boundary in the middle of a character. The resulting "dangling byte" is rendered as U+FFFD and the byte sequence itself is not valid UTF-8. This can interfere with the validity of a given text file. Unlike many legacy character encodings, UTF-8 is highly patterned, so the the longest broken character sequence that can result from mid-character truncation is one character. By contrast, in many legacy encodings, a file or document containing a mid-character truncated string can be wholly changed or rendered unintelligible after that point.</p>
-		<table class=truncExample>
-		<tr>
-			<th>Script</th>
-			<th>Truncated Length<br>(code points)</th>
-			<th>Avg. Bytes/Code Point</th>
-			<th>Truncated Text<br>Byte Values</th>
-		</tr>
-		<tr>
-			<td rowspan=2>ASCII
-			<td rowspan=2>40</td>
-			<td rowspan=2>1</td>
-			<td><kbd>In the loveliest town of all, where the </kbd></td>
-		</tr>
-		<tr>
-			<td><code>49 6E 20 74 68 65 20 6C 6F 76 65 6C 69 65 73 74 20 74 6F 77 6E 20 6F 66 20 61 6C 6C 2C 20 77 68 65 72 65 20 74 68 65 20</code></td>
-		</tr>
-		<tr>
-			<td rowspan=2>Cyrillic</td>
-			<td rowspan=2>22</td>
-			<td rowspan=2>2</td>
-			<td><kbd>В самом прекрасном го�</kbd></td>
-	    </tr>
-	    <tr>
-			<td><code>D0 92 20 D1 81 D0 B0 D0 BC D0 BE D0 BC 20 D0 BF D1 80 D0 B5 D0 BA D1 80 D0 B0 D1 81 D0 BD D0 BE D0 BC 20 D0 B3 D0 BE <span style="color:red">D0</span></code></td>
-	    </tr>
-		<tr>
-			<td rowspan=2>Han</td>
-			<td rowspan=2>14</td>
-			<td rowspan=2>3</td>
-			<td><kbd lang="zh">在最美丽的城镇，那里的房屋�</kbd></td>
-		</tr>
-		<tr>
-			<td><code>E5 9C A8 E6 9C 80 E7 BE 8E E4 B8 BD E7 9A 84 E5 9F 8E E9 95 87 EF BC 8C E9 82 A3 E9 87 8C E7 9A 84 E6 88 BF E5 B1 8B <span style="color:red">E5</span></code></td>
-	    </tr>
-		<tr>
-			<td rowspan=2>Emoji</td>
-			<td rowspan=2>10
-			<td rowspan=2>4</td>
-			<td><kbd>🙊🙁😢😠😧😎😽😉😄😮</kbd></td>
-		</tr>
-		<tr>
-			<td><code>F0 9F 99 8A F0 9F 99 81 F0 9F 98 A2 F0 9F 98 A0 F0 9F 98 A7 F0 9F 98 8E F0 9F 98 BD F0 9F 98 89 F0 9F 98 84 F0 9F 98 AE</code></td>
-		</tr>
-		</table>
     </aside>
 
-
-	<div class="req" id="char_trunc_units">
-	<p class="advisement">Specifications that limit the length of a string MUST specify which type of unit (extended grapheme clusters, Unicode code points, or code units) the length limit uses.</p>
-	</div>
-
+ 
+    
 	<div class="req" id="char_trunc_unit_rec">
 	<p class="advisement">Specifications that limit the length of a string SHOULD specify the length in terms of Unicode code points.</p>
```

Comment:
```suggestion
	<p class="advisement">Specifications that limit the maximum allowable length of a string SHOULD specify the length in terms of Unicode code points.</p>
```

This may help to make it clear that we're talking about container lengths here, rather than truncation points.